backend/czi_hosted/data_common/matrix_loader.py

```diff
@@ -152,6 +155,21 @@ def data_adaptor(self, url_dataroot, location, app_config):
                 cache_item = info.cache_item
 
             if data_adaptor is None:
+                if app_config.server_config.data_locator__api_base:
+                    if url_dataroot == "e":  # only pull dataset identification information from the portal for /e/ route
+                        headers = {"Content-Type": "application/json"}
+                        curr_url = f"{app_config.server_config.get_web_base_url()}/{url_dataroot}/{dataset}"
+                        response = requests.get(
```

Comment:
this can throw exceptions in many situations [docs](https://docs.python-requests.org/en/latest/api/#exceptions). I"m not sure that you want it to throw up the stack in the case where the Explorer API gateway can't talk to the Portal API service, unless all requests for the dataset are invalid when the API gateway is unreachable (or maybe you do - would be good to document the error path, as these network failures will definitely occur).